core/domain/stats_domain.py

```diff
@@ -63,13 +66,119 @@
 MAX_ANSWER_DETAILS_BYTE_SIZE = 10000
 
 
+class SubmittedAnswerDict(TypedDict, total=False):
```

Comment:
Why total=False -- what keys may be missing?

Reply:
rule_spec_str and answer_str may be missing